clang/lib/DPCT/APINamesThrust.inc

```diff
@@ -144,7 +144,7 @@ CONDITIONAL_FACTORY_ENTRY(
                  ARG(0), ARG(1), ARG(3), ARG(4), ARG(2))),
         CALL_FACTORY_ENTRY("thrust::transform_reduce",
                            CALL("std::transform_reduce",
-                                ARG("oneapi::dpl::execution::seq"), ARG(0),
+                                ARG("oneapi::dpl::execution::par_noseq"), ARG(0),
```

Comment:
`oneapi::dpl::execution::par_noseq` should be `oneapi::dpl::execution::par_unseq` throughout this PR. The oneDPL execution policies that do not offload to devices have the same names as standard C++ execution policies.

Reply:
I will update this PR to confirm if it can pass the alloy test.

Reply:
@timmiesmith using `oneapi::dpl::execution::par_unseq` will result feature test regression:
```
===============Failed case(s) ==========================
thrust-binary-search COMPFAIL 
thrust-copy COMPFAIL 
thrust-transform-if COMPFAIL 
thrust-gather COMPFAIL 
thrust-gather_if COMPFAIL 
thrust-scatter COMPFAIL 
thrust-unique_by_key_copy COMPFAIL 
thrust-unique_by_key COMPFAIL 
thrust_minmax_element COMPFAIL 
thrust_partition COMPFAIL 
thrust_is_sorted COMPFAIL 
thrust_replace_copy COMPFAIL 
thrust_scatter_if COMPFAIL 
thrust_stable_partition COMPFAIL 
thrust_random_type COMPFAIL 
thrust_uninitialized_fill_n COMPFAIL 
thrust_swap_ranges COMPFAIL 
thrust_reverse COMPFAIL 
thrust_transform_inclusive_scan COMPFAIL 
thrust_remove_copy COMPFAIL 
thrust_unique_copy COMPFAIL 
thrust_transform_exclusive_scan COMPFAIL 
thrust_set_difference COMPFAIL 
thrust_set_difference_by_key COMPFAIL 
thrust_set_intersection_by_key COMPFAIL 
thrust_stable_sort COMPFAIL 
thrust_tabulate COMPFAIL 
thrust_for_each_n COMPFAIL 
```
More detail, pls refer to [compile error log file](https://pipelines.actions.githubusercontent.com/serviceHosts/238081d5-d470-4351-bec7-9f0b79b593ac/_apis/pipelines/1/runs/9115/signedlogcontent/34?urlExpires=2023-08-17T09%3A14%3A15.9845303Z&urlSigningMethod=HMACV1&urlSignature=Yzy54fMMli%2BgGTXmdj%2BV3IAJ2tZOFPTGhpABXZBhfDo%3D)